### ISSUE TITLE
Music Files: No paging on playlist searches

### ISSUE THREAD

**Issue by Moonbase59** [AzuraCast/AzuraCast#7229]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

Rolling Release #35eca37 (2024-06-21 14:26)

### What happened?

Sometimes you wish to quickly look up what’s in a playlist, to possibly edit the tracks. Here is a playlist with 13 entries:

![Auswahl_440](https://github.com/AzuraCast/AzuraCast/assets/3706922/e563cd09-a4ee-4e0d-bd35-1385756d434d)

I now invoke a playlist search by clicking on the `13` and _Music Files_ only shows the first 10, without paging—only page 1 is offered.

![Music Files-Playlist Selection-No Paging](https://github.com/AzuraCast/AzuraCast/assets/3706922/1891cf74-b28b-4672-873b-14bc0078f793)

**Comment by BusterNeece** [comment]
@Moonbase59 I can't reproduce this one locally. There is a momentary delay if you click the number on the playlists page as it transitions over to the media page and then reloads it to run a search, but on my end it's properly paginated without issue. Not sure why it wouldn't be on your end.

**Comment by Moonbase59** [comment]
@BusterNeece Sorry, seems I was mistaken. It’s for some reason _not_ the paging, but several playlists schow _more entries_ than they seem to have!?

Here is a short video to show:

https://github.com/AzuraCast/AzuraCast/assets/3706922/956f8274-4f2e-4e32-8065-8c48893e53e6

**Comment by BusterNeece** [comment]
@Moonbase59 Are they sequential? That should be the only playlist type that can hold a single song listed multiple times inside the playlist.

**Comment by Moonbase59** [comment]
No, normal song-based, no schedule, no advanced:

![Auswahl_452](https://github.com/AzuraCast/AzuraCast/assets/3706922/d2fdf338-8570-4df2-b458-c247518b8aff)

**Comment by Moonbase59** [comment]
Is there some way to find out if I specified a whole folder (`media/Other/Test/Current` possibly) when originally creating the playlist? This could be, but since adding/removing files from that folder didn’t seem to add them, I might have added them manually in _Media Files_. Can’t really remember, just an idea.

**Comment by BusterNeece** [comment]
@Moonbase59 If you look at the parent directory and the directory _itself_ shows up as being in the playlist, then that means it's set up to automatically put everything underneath it into the playlist. If not, then it's just individual items inside it.

**Comment by Moonbase59** [comment]
Thanks! That’s a good thing to know!

Interesting: When I export that playlist via the API, I get 13, and duplicates:
```
Other/Test/Current/Beatles, The - Strawberry Fields Forever.flac
Other/Test/Current/Beatles, The - Strawberry Fields Forever.flac
Other/Jingles/test-5-15-5-15-5s.mp3
Other/Test/Current/Darkness on Demand - Quicksand.ogg
Other/Test/Current/ella_henderson_rudimental_-_alibi_feat._rudimental.mp3
Other/Test/Current/Wonder, Stevie - Another Star.m4a
Other/Test/Current/Queen - Bohemian Rhapsody.flac
Other/Test/Current/Beatles, The - Strawberry Fields Forever.flac
Other/Test/Current/TLC - Waterfalls.mp3
Other/Test/Current/Beatles, The - Strawberry Fields Forever.flac
Other/Test/Current/Stürmer, Christina - Ich lebe.mp3
Other/Test/Current/robbie_williams_-_angels.mp3
Other/Test/Current/Electric Light Orchestra - Don't Bring Me Down.mp3
```

**Comment by Moonbase59** [comment]
How can we ensure such discrepancies don’t happen? Or at least remove the dups?

**Comment by BusterNeece** [comment]
@Moonbase59 Was that playlist _ever_ sequential by chance? It shouldn't be possible for that to happen, but it's certainly possible if you assign media into a sequential playlist, and then switch it.

**Comment by Moonbase59** [comment]
Certainly not. I always wanted to play with that feature but never yet did.

**Comment by Moonbase59** [comment]
But I _will_ surely have clicked the Beatles song and tried to add it _before_ the "File processing" indicator vanished. Because I thought I could add to the playlist before that was finished.

**Comment by Moonbase59** [comment]
So apparently "user error" and I just caught "the hole in between".

I wonder if we could/should prevent such a thing happening on non-sequential playlists maybe? Like a with a message "File is already in playlist" or the like?

**Comment by BusterNeece** [comment]
@Moonbase59 It really shouldn't be happening and as far as I can tell it shouldn't happen in current code, but in this case you can certainly fix it just by removing the files from all playlists and then adding them back to that playlist.

**Comment by Moonbase59** [comment]
From a user perspective: Is there any easy way _within AzuraCast_ to find which files are duplicates in a playlist? Looking through my playlists, I seem to have several that have more entries than search shows.

**Comment by BusterNeece** [comment]
@Moonbase59 Not currently. Just trying to fix this isolated issue with you. Not confident this is widespread.

**Comment by Moonbase59** [comment]
There must be some bug in that: Beatles shown as belonging to playlist, I try to add it _again_ because the playlists have no checkmark, and the result is all Betales entries _removed_!

https://github.com/AzuraCast/AzuraCast/assets/3706922/47f1d5b1-4bb7-4b9e-819f-0eb9c071fca6

**Comment by Moonbase59** [comment]
Before:
```
Other/Test/Current/Beatles, The - Strawberry Fields Forever.flac
Other/Test/Current/Beatles, The - Strawberry Fields Forever.flac
Other/Jingles/test-5-15-5-15-5s.mp3
Other/Test/Current/Darkness on Demand - Quicksand.ogg
Other/Test/Current/ella_henderson_rudimental_-_alibi_feat._rudimental.mp3
Other/Test/Current/Wonder, Stevie - Another Star.m4a
Other/Test/Current/Queen - Bohemian Rhapsody.flac
Other/Test/Current/Beatles, The - Strawberry Fields Forever.flac
Other/Test/Current/TLC - Waterfalls.mp3
Other/Test/Current/Beatles, The - Strawberry Fields Forever.flac
Other/Test/Current/Stürmer, Christina - Ich lebe.mp3
Other/Test/Current/robbie_williams_-_angels.mp3
Other/Test/Current/Electric Light Orchestra - Don't Bring Me Down.mp3
```

After:
```
Other/Jingles/test-5-15-5-15-5s.mp3
Other/Test/Current/Darkness on Demand - Quicksand.ogg
Other/Test/Current/ella_henderson_rudimental_-_alibi_feat._rudimental.mp3
Other/Test/Current/Wonder, Stevie - Another Star.m4a
Other/Test/Current/Queen - Bohemian Rhapsody.flac
Other/Test/Current/TLC - Waterfalls.mp3
Other/Test/Current/Stürmer, Christina - Ich lebe.mp3
Other/Test/Current/robbie_williams_-_angels.mp3
Other/Test/Current/Electric Light Orchestra - Don't Bring Me Down.mp3
```

**Comment by BusterNeece** [comment]
@Moonbase59 That's a product of the weird duplication thing you have going on. It iterates though all media in a playlist and basically, this is what happens if it finds it twice. Should set it on the second attempt.

Given how long your station's been running, it's impossible to know if it wound up in this duplicate-playlist situation because of current code or any of the thousands of changes that have happened since you spun the install up. We'd need to see it happen on a new install to indicate it happens with current code.

**Comment by Moonbase59** [comment]
I could create a new playlist, think that’s be good enough?

**Comment by BusterNeece** [comment]
@Moonbase59 Possibly. Please check other issues btw, I'm waiting on you to resolve another thing.

**Comment by Moonbase59** [comment]
Seems to work alright so far, except what confuses me is when the song _is_ already in on or mor playlists, shouldn’t the "long checkmark dropdown" indicate that so one could take it _off_ one or more playlists?

**Comment by Moonbase59** [comment]
Closing this to unclutter the issues.
Maybe I’ll make another issue about the playlist checkmarks.
